gnucash/gnome-utils/gnc-main-window.cpp

```diff
@@ -118,6 +119,7 @@ enum
 #define GNC_PREF_TAB_OPEN_ADJACENT    "tab-open-adjacent"
 
 #define GNC_MAIN_WINDOW_NAME "GncMainWindow"
+#define GNC_MAIN_WINDOW_MAX_NUMBER 10
```

Comment:
From a cursory read through the rest of the code it looks like the list of windows to select from the Windows menu is limited to 10 and this define is to keep that same restriction in your added code.
Should we really keep it ? Or can we just rewrite the code to allow an arbitrarily long list of open windows to select ?

Reply:
I did this as I am only using a single digit to identify the already added window menu entry action names like Window0Action to Window9Action. It looks like I will have to create and add the menu items and can use two digits so it would support 99 windows which I think should be adequate and that would do away with the define.

Reply:
> Or can we just rewrite the code to allow an arbitrarily long list of open windows to select ?

Let's not. That list is displayed in the Windows menu and AFAIK there's no scrollbar available in GTK menus. It looks to me like our nominal 800x600 minimum display size will handle only around 25 lines of menu and then only if the window is all the way at the top of the screen.

I also don't recall anyone ever complaining that they couldn't create enough windows.

> so it would support 99 windows which I think should be adequate and that would do away with the define.

No, it would just change the define to 100. You still need the number to test the index against.

That aside `static unsigned gnc_main_window_max_number{10};` is better than a macro unless you need the value to initialize a static array, which you don't. Since it's a C++ file `constexpr` is better than `static`, and you *can* use `constexpr` values to initialize static arrays.

Reply:
OK, I will leave this at 10 and change to use a `constexpr`. Interestingly even though there are only 10 entries [here](https://github.com/Gnucash/gnucash/blob/3c306eae658e3b4645eb51fc33a45205eaabf4dd/gnucash/gnome-utils/gnc-main-window.cpp#L476) you can still keep creating new windows, I stopped at 12. I suppose `gnc_main_window_cmd_window_move_page` should check the `active_windows` list length and stop at 10, should a dialog be presented advising max number of windows open?

Reply:
Maybe a message box saying that the menu max has been exceeded and more windows won't be added to the Window menu.  Be sure to wrap it in `ifndef MAC_INTEGRATION`, macOS manages the Windows menu on its own.